cli/cmd/get_cmd.go

```diff
@@ -2,47 +2,48 @@ package cmd
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
 	"github.com/kubeshop/tracetest/cli/parameters"
 	"github.com/kubeshop/tracetest/cli/pkg/resourcemanager"
 	"github.com/spf13/cobra"
 )
 
-var getParams = &parameters.ResourceIdParams{}
-
-var getCmd = &cobra.Command{
-	GroupID: cmdGroupResources.ID,
-	Use:     fmt.Sprintf("get %s", strings.Join(parameters.ValidResources, "|")),
-	Short:   "Get resource",
-	Long:    "Get a resource from your Tracetest server",
-	PreRun:  setupCommand(),
-	Run: WithResourceMiddleware(func(_ *cobra.Command, args []string) (string, error) {
-		resourceType := args[0]
-		ctx := context.Background()
-
-		resourceClient, err := resources.Get(resourceType)
-		if err != nil {
-			return "", err
-		}
-
-		resultFormat, err := resourcemanager.Formats.Get(output, "yaml")
-		if err != nil {
-			return "", err
-		}
-
-		result, err := resourceClient.Get(ctx, getParams.ResourceID, resultFormat)
-		if err != nil {
-			return "", err
-		}
-
-		return result, nil
-	}, getParams),
-	PostRun: teardownCommand,
-}
+var (
+	getParams = &parameters.ResourceIdParams{}
+	getCmd    *cobra.Command
+)
 
 func init() {
+	getCmd = &cobra.Command{
+		GroupID: cmdGroupResources.ID,
+		Use:     "get " + resourceList(),
+		Short:   "Get resource",
+		Long:    "Get a resource from your Tracetest server",
+		PreRun:  setupCommand(),
+		Run: WithResourceMiddleware(func(_ *cobra.Command, args []string) (string, error) {
+			resourceType := args[0]
```

Comment:
Can we get a panic here if the user runs the command just as `tracetest get` ?
Should we validate the `args` size and print the help on this case?

Reply:
[WithResourceMiddleware](https://github.com/kubeshop/tracetest/blob/f112ab7e887afb558435116cc6017337d9a59e23/cli/cmd/middleware.go#L65) validates that the resource is set before getting to this function. Here's an example of running `list` without passing a resource: 

![Screenshot 2023-06-28 at 15 34 46](https://github.com/kubeshop/tracetest/assets/314548/ac7d6d0f-5341-474b-8fbd-2e6203422e29)

Reply:
It makes sense. My fear about accessing indexes in languages that throw out-of-index errors and friends is that sooner or later, we can receive that error (usually on dev mode when we coded something wrong by mistake xD).

One thing that we can think about in the future is that since the `WithResourceMiddleware` is tailored to the resources API, we could unpack the first argument and change the interface to `func(_ *cobra.Command, resourceType string, args []string) (string, error)` and allow `args` to have only the args for that command in fact.

Reply:
> sooner or later, we can receive that error
This is true. Assuming that the middleware will always work perfectly might not be the safest approach. This is actually being parsed to the `resourceParams.ResourceName` property, so I replaced the calls to `args[0]` to `resourceParams.ResourceName`.

What do you think @danielbdias? It doesn't guarantee success, but at least it doesn't panic

Reply:
It sounds good to me!